lexers/e/easytrieve.go

```diff
@@ -0,0 +1,135 @@
+package e
+
+import (
+	"regexp"
+	"strings"
+
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+var (
+	easytrieveAnalyserCommetLineRe  = regexp.MustCompile(`^\s*\*`)
+	easytrieveAnalyserMacroHeaderRe = regexp.MustCompile(`\s*MACRO`)
+)
+
+// Easytrieve lexer.
+var Easytrieve = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "Easytrieve",
+		Aliases:   []string{"easytrieve"},
+		Filenames: []string{"*.ezt", "*.mac"},
+		MimeTypes: []string{"text/x-easytrieve"},
+	},
+	Rules{
+		"root": {},
+	},
+).SetAnalyser(func(text string) float32 {
+	// Perform a structural analysis for basic Easytrieve constructs.
+	var (
+		result           float32
+		hasEndProc       bool
+		hasHeaderComment bool
+		hasFile          bool
+		hasJob           bool
+		hasProc          bool
+		hasParm          bool
+		hasReport        bool
+	)
+
+	lines := strings.Split(text, "\n")
+
+	// Remove possible empty lines and header comments.
+	for range lines {
+		if len(strings.TrimSpace(lines[0])) > 0 && !easytrieveAnalyserCommetLineRe.MatchString(lines[0]) {
+			break
+		}
+
+		if easytrieveAnalyserCommetLineRe.MatchString(text) {
+			hasHeaderComment = true
+		}
+
+		lines = lines[1:]
+	}
+
+	if easytrieveAnalyserMacroHeaderRe.MatchString(lines[0]) {
+		// Looks like an Easytrieve macro.
+		result += 0.4
+
+		if hasHeaderComment {
+			result += 0.4
+		}
+
+		return result
+	}
+
+	// Scan the source for lines starting with indicators.
+	for _, line := range lines {
+		splitted := strings.Fields(line)
+
+		if len(splitted) < 2 {
+			continue
+		}
+
+		if !hasReport && !hasJob && !hasFile && !hasParm && splitted[0] == "PARM" {
+			hasParm = true
+		}
+
+		if !hasReport && !hasJob && !hasFile && splitted[0] == "FILE" {
+			hasFile = true
+		}
+
+		if !hasReport && !hasJob && splitted[0] == "JOB" {
+			hasJob = true
+		}
+
```

Comment:
The code does not fully take into account, the `elif` statement at https://github.com/pygments/pygments/blob/master/pygments/lexers/scripting.py#L1114. One way to solve this is to add another condition here:
```go
if !hasReport && !hasJob {
    continue
}
```

Reply:
This refactoring was crazy :P